src/fides/cli/utils.py

```diff
@@ -99,35 +106,129 @@ def handle_cli_response(
     return response
 
 
+def create_config_file(ctx: click.Context, fides_directory_location: str = ".") -> str:
+    """
+    Creates the .fides/fides.toml file and initializes it, if it doesn't exist.
+
+    Returns the config_path if successful
+    """
+    fides_dir_name = ".fides"
+    fides_dir_path = f"{fides_directory_location}/{fides_dir_name}"
+    config_file_name = "fides.toml"
+    config_path = f"{fides_dir_path}/{config_file_name}"
+    config = ctx.obj["CONFIG"]
+
+    included_values = {
+        "database": {
+            "server",
+            "user",
+            "password",
+            "port",
+            "db",
+        },
+        "logging": {
+            "level",
+            "destination",
+            "serialization",
+        },
+        "cli": {"server_protocol", "server_host", "server_port"},
+    }
+
+    # create the .fides dir if it doesn't exist
+    if not os.path.exists(fides_dir_path):
+        os.mkdir(fides_dir_path)
+        click.echo(f"Created a '{fides_dir_path}' directory.")
+    else:
+        click.echo(f"Directory '{fides_dir_path}' already exists.")
+
+    # create a fides.toml config file if it doesn't exist
+    if not os.path.isfile(config_path):
+        with open(config_path, "w", encoding="utf-8") as config_file:
+            config_dict = config.dict(include=included_values)
+            toml.dump(config_dict, config_file)
+        click.echo(f"Created a fides config file: {config_path}")
+    else:
+        click.echo(f"Configuration file already exists: {config_path}")
+
+    click.echo("To learn more about configuring fides, see:")
+    click.echo("\thttps://ethyca.github.io/fides/installation/configuration/")
+
+    return config_path
+
+
+def is_user_registered(ctx: click.Context) -> bool:
+    """
+    Send a request to the API server, and determine if a registration is already present.
+    """
+
+    response = get(f"{ctx.obj['CONFIG'].cli.server_url}{V1_URL_PREFIX}{REGISTRATION}")
+    return response.json()["opt_in"]
+
+
+def register_user(ctx: click.Context, email: str, organization: str) -> None:
+    """
+    Create a new registration record in the database.
+    """
+
+    put(
+        f"{ctx.obj['CONFIG'].cli.server_url}{V1_URL_PREFIX}{REGISTRATION}",
+        json={
+            "analytics_id": ctx.obj["CONFIG"].cli.analytics_id,
+            "opt_in": True,
+            "user_email": email,
+            "user_organization": organization,
+        },
+    )
+
+
 def check_and_update_analytics_config(ctx: click.Context, config_path: str) -> None:
     """
     Ensure the analytics-related config is present. If not,
     prompt the user to opt-in to analytics and/or update the
     config file with their preferences.
     """
 
+    # TODO: check if server is connected
+    is_server_connected = True
+    should_attempt_registration = is_server_connected and not is_user_registered(ctx)
```

Comment:
@ThomasLaPiana this is the one part where I ran out of steam - I just want this function to transparently handle both cases:
* CLI is connected to a server
* CLI is not connected to a server

I didn't see a quick & clean way to do that and I figured there probably was one, so I skipped over it... can you finish up? 

I've designed the logic branches below to work accordingly, and this function can be called multiple times by the same CLI and then the _first_ time it connects to a server, it'll prompt for consent again to register the server, etc.

Reply:
added